src/wallet/wallet.h

```diff
@@ -1059,7 +1059,7 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
                 // are empty. This covers transactions that have no Sprout or Sapling data
                 // (i.e. are purely transparent), as well as shielding and unshielding
                 // transactions in which we only have transparent addresses involved.
-                if (!(wtx.mapSproutNoteData.empty() && wtx.mapSaplingNoteData.empty())) {
+                if (!(wtx.mapSproutNoteData.empty() && wtx.mapSaplingNoteData.empty() && wtx.mapOrchardActionData.empty())) {
```

Comment:
Undo this change. We serialize `CWalletTx`s that have Sprout or Sapling note data, because that note data is mutated as the chain tip changes to update the note witnesses. Orchard witnesses are tracked at the Orchard wallet level, so we need to serialize _that_ in this function (and ensure that it is exactly in sync with the best chain point).
```suggestion
                if (!(wtx.mapSproutNoteData.empty() && wtx.mapSaplingNoteData.empty())) {
```